src/typescript/frontend/src/components/pages/launch-emojicoin/hooks/use-register-market.ts

```diff
@@ -33,9 +33,20 @@ export const useRegisterMarket = () => {
   const { data: gasResult } = useQuery({
     queryKey: ["register-market-cost", numMarkets, account?.address, emojiBytes],
     queryFn: async () => {
-      const publicKey = new Ed25519PublicKey(
-        typeof account!.publicKey === "string" ? account!.publicKey : account!.publicKey[0]
-      );
+      let publicKey: Ed25519PublicKey;
+      try {
+        publicKey = new Ed25519PublicKey(
+          typeof account!.publicKey === "string" ? account!.publicKey : account!.publicKey[0]
+        );
+      } catch (_) {
+        return {
+          error: true,
+          data: {
+            amount: 0,
+            unitPrice: 0,
+          }
```

Comment:
Yes. It seems that Aptos Connect public keys are not ed25519 public keys... I was unable to find reliable information on how to handle those.

Reply:
Commenting here to remind you that you can try a few different things:

1. You can try using a placeholder public key, since I believe that simulations don't verify that the public key matches the authentication key. So it can possibly be anything, you will need to test this
2. To find the public key type, try to create it with different public key types with the SDK and then match the resulting account address to the public key. I don't think this is 100% reliable if they have rotated their key. You could always do a check for this (although I think this is excessively robust/unnecessary)